### PR TITLE
RHC Changes - Adding new API Version

### PR DESCRIPTION
**Pull request by sankalpsoni**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by veronicagg** [comment]
@sankalpsoni If you're making a new API version based on an existing API version, we ask you to please consider making a commit with the older version and another one with the changes, this helps with the review, please see our contribution guidelines https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md#filenames-and-folder-structure.

**Comment by veronicagg** [comment]
@sankalpsoni have you taken a look at https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md#filenames-and-folder-structure, is not flattening intentional in this case?

File: arm-resourcehealth/2017-07-01/swagger/resourcehealth.json
Comment (veronicagg) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: should "its" be "it's" ?

File: arm-resourcehealth/2017-07-01/swagger/resourcehealth.json
Comment (veronicagg) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: have these properties been added to the service in this new API version?

Comment (veronicagg) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: should this property be an enum? https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-enum

Comment (veronicagg) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: is this a read-only property?

File: arm-resourcehealth/2017-07-01/swagger/resourcehealth.json
Comment (veronicagg) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: should this property be an enum? is it read-only?

File: arm-resourcehealth/2017-07-01/swagger/resourcehealth.json
Comment (veronicagg) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: similar questions as previous properties

File: arm-resourcehealth/2017-07-01/swagger/resourcehealth.json
Comment (veronicagg) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: formatting: please fix indenting.
Is this a read-only property?

**Review by veronicagg — CHANGES_REQUESTED** [review]

**Comment by veronicagg** [comment]
@ravbhatnagar please let me know if you want to sign-off on this one, based on my comparison withe the previous API version, there are only a few added properties where I posted questions above. Rest of the spec looks the same as previous API version. thanks!

File: arm-resourcehealth/2017-07-01/swagger/resourcehealth.json
Comment (ravbhatnagar) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: If this has a known set of values, please make it an enum. It will be better from usability POV

File: arm-resourcehealth/2017-07-01/swagger/resourcehealth.json
Comment (ravbhatnagar) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: Yes, looks like a candidate for enum

File: arm-resourcehealth/2017-07-01/swagger/resourcehealth.json
Comment (ravbhatnagar) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: +1 to Veronica's comment

**Comment by ravbhatnagar** [comment]
@veronicagg - Took a look at the property changes. Same concerns as yours. Else looks good from ARM perspective. At your discretion for merging

**Comment by veronicagg** [comment]
Based on @sankalpsoni offline response "There are other similar properties which are string in old version. And would like to stick with “string” in the new version as well. "
Please evaluate this on the next api version you create. 

@sankalpsoni Please fix the formatting issue in line https://github.com/Azure/azure-rest-api-specs/pull/1375/files#diff-07c6cc2814d7fcc094711628d774e0b3R339 and let me know if the properties are "read-only" and you'll be updating them or not. thanks!

**Comment by fearthecowboy** [comment]
ATTENTION REQUIRED.
-------------

We have changed the `default` branch in the `azure-rest-api-specs` repository to use the branch labeled `current`.

This PR is opened against the `master` branch.

You have two options: 
1. Keep this PR open, and complete the review process as normal, and once this PR is merged against the `master` branch, we (@olydis & @fearthecowboy)  will create a PR against the current branch and make any updates required to work in the newly reorganized default branch.  

2. Close this PR and create a new one. That would mean pulling in the changes from the new default branch `current` moving any files and switching to the configuration-based layout.  (You are more than welcome to choose option 2, but I don't expect anyone to do that)

Either way, the work you are currently doing against the previous default branch `master` will be the last PR against that branch that we will accept and all new PRs will go against the new default branch  `current`.

If you have any questions or concerns, ping @fearthecowboy or @olydis (or go ask on the `#swagger` channel in slack) ASAP and we can help you thru the process.

File: arm-resourcehealth/2017-07-01/swagger/resourcehealth.json
Comment (sankalpsoni) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: Yes. Modifying

File: arm-resourcehealth/2017-07-01/swagger/resourcehealth.json
Comment (sankalpsoni) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: Yes, these properties are added to this new API version. We will make them enum in future API version. These and lot of old properties are strings, and we want to be consistent.

File: arm-resourcehealth/2017-07-01/swagger/resourcehealth.json
Comment (sankalpsoni) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: We will make them enum in future API version. These and lot of old properties are strings, and we want to be consistent.

File: arm-resourcehealth/2017-07-01/swagger/resourcehealth.json
Comment (sankalpsoni) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: Done. 
Yes, these and other properties which are present since version 1 are read only. 
Is there any special guidance for read-only properties? If so, we would be glad to get to them in next api-version.

File: arm-resourcehealth/2017-07-01/swagger/resourcehealth.json
Comment (sankalpsoni) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: Yes, these and other properties which are present since version 1 are read only. Is there any special guidance for read-only properties? If so, we would be glad to get to them in next api-version.
--

**Review by sankalpsoni — COMMENTED** [review]
Updated review with comments

File: arm-resourcehealth/2017-07-01/swagger/resourcehealth.json
Comment (veronicagg) [inline arm-resourcehealth/2017-07-01/swagger/resourcehealth.json]: indenting still looks off.

**Review by veronicagg — CHANGES_REQUESTED** [review]
left one comment as formatting still looks off. 
Regarding read only properties, yes, you can add a property "readOnly": true, to indicate this, see https://swagger.io/specification/

readOnly | boolean | Relevant only for Schema "properties" definitions. Declares the property as "read only". This means that it MAY be sent as part of a response but MUST NOT be sent as part of the request. Properties marked as readOnly being true SHOULD NOT be in the required list of the defined schema. Default value is false
-- | -- | --

Please note the comment about the changes in the repo structure, would you like to continue with this PR or update it to be against the "current" branch?

**Comment by veronicagg** [comment]
any updates?

**Comment by sankalpsoni** [comment]
Hi Veronica - fixed the indentation issue.
About “readOnly”: true -  this is applicable on other properties as well, and to be consistent - it needs to be added to all properties. We will evaluate and edit in the next API version.

**Comment by sankalpsoni** [comment]
As given an option by @fearthecowboy, I would keep this PR open and let the him/his team merge against current.

**Comment by veronicagg** [comment]
@sankalpsoni that's ok, regarding the read-only properties. Regarding your last comment, it looks that you've closed this PR, so I can't really approve it, what's the plan on it?

**Comment by sankalpsoni** [comment]
Hi - I have reopened it. My mistake :(

**Review by veronicagg — APPROVED** [review]

**Comment by veronicagg** [comment]
@fearthecowboy @olydis this PR can now be merged and transferred to `current` branch, for some reason I'm not able to merge this one into `master` myself.

**Pull request merged**
